app/components/input-payment/input-payment.tsx

```diff
@@ -92,12 +92,12 @@ export const InputPayment: ComponentType = ({
   prefCurrency,
   nextPrefCurrency,
 }: InputPaymentDataInjectedProps) => {
-  const endByDot = (s: string) => s.match(/^[0-9]*\.{1}$/)
-
   const [amount, setAmount] = React.useState(initAmount)
   const [input, setInput] = React.useState("")
-  const [appendDot, setAppendDot] = React.useState(false)
 
+  const handleChange = (text) => {
+    setInput(currency === CurrencyType.USD || currency === CurrencyType.BTC ? toCurrency(text) : text)
```

Comment:
I think instead of doing a check here you could pass `currency` to the function and keep more logic in the `toCurrency` function.